appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryPage.java

```diff
@@ -931,8 +936,13 @@ public void onSuccess(Long id) {
                           submitReport.setEnabled(false);
                         }
                     };
-                  Ode.getInstance().getGalleryService().addAppReport(app, reportText.getText(),
-                      reportClickCallback);
+                  if (reportText.getText().replaceAll(" ", "").length() > 0){
```

Comment:
what does `reportText.getText().replaceAll(" ", "").length() > 0` do?
I understand that it gets some text, replaces all spaces, and then checks if it's not empty?
Can the code be a bit more semantical? maybe using methods like trim() and isEmpty? I am having a hard time trying to figure out what the clause actually is.